sagemaker-python-sdk/tensorflow_script_mode_training_and_serving/tensorflow_script_mode_training_and_serving.ipynb

```diff
@@ -0,0 +1,239 @@
+    "                             framework_version='1.12',\n",
+    "                             py_version = 'py3',\n",
+    "                             distributions = {'parameter_server': {'enabled': True}},\n",
+    "                             base_job_name='test-tf')\n",
+    "\n",
+    "mnist_estimator.fit(training_data_uri)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To start a training job, we call `estimator.fit(training_data_uri)`,\n",
+    "\n",
+    "You can pass a dictionary or a simply a S3 location to the ``fit`` method. If a dictionary is used the keys, named **channels**, \n",
+    "have values pointing to the data location. `estimator.fit(training_data_uri)` downloads the TensorFlow container with TensorFlow Python 3, GPU version to the SageMaker training job instances. If a S3 location is used the estimator will create a default **channel** named ``'training'`` which points to this S3 location.\n",
+    "When training starts, the TensorFlow container executes **mnist.py**, passing `hyperparameters` which we don't have any in our case and `model_dir` as script arguments, executing the example as follows:\n",
+    "```bash\n",
+    "python mnist.py --model_dir s3://<DEFAULT_BUCKET>/<TRAINING_JOB>\n",
+    "```\n",
+    "In the end of the training, the training job will upload the saved model for TensorFlow serving."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "# Deploy the trained model to prepare for predictions\n",
+    "\n",
+    "The deploy() method creates an SageMaker model which is then used to create an endpoint to serve prediction requests in real time. In this tutorial the endpoint is created with the TensorFlow serving container which is used by default for a script mode ``Estimator``. This serving container runs an implementation of a web server which is compatible with SageMaker hosting protocol. It pareses the ``/invocation`` calls then passes it to the backend TensorFlow serving server. The [Using your own inference code]() document explains how inference images are run in SageMaker."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "predictor = mnist_estimator.deploy(initial_instance_count=1, instance_type='ml.p2.8xlarge')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Invoking the endpoint\n",
+    "\n",
+    "Let's download the training data and use that as input for inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "!aws s3 cp s3://sagemaker-sample-data-us-west-2/tensorflow/mnist/train_data.npy train_data.npy\n",
+    "!aws s3 cp s3://sagemaker-sample-data-us-west-2/tensorflow/mnist/train_labels.npy train_labels.npy\n",
```

Comment:
the region part should be dynamic. also you might need to put `--region` in the command to make sure it works everywhere (it might be easier to use https for GovCloud)

Reply:
not sure if the cli supports https. I will add the region here.